tests/chainer_tests/links_tests/connection_tests/test_deconvolution_nd.py

```diff
@@ -0,0 +1,102 @@
+import unittest
+
+import numpy
+
+import chainer
+from chainer import cuda
+from chainer import gradient_check
+from chainer import initializers
+from chainer.links import deconvolution_nd
+from chainer import testing
+from chainer.testing import attr
+from chainer.testing import condition
+from chainer.testing import parameterize
+from chainer.utils import conv
+
+
+@parameterize(*testing.product({
+    'dims': [(5, 4, 3), (4, 3), (3,)],
+    'nobias': [True, False],
+    'dtype': [numpy.float32],   # TODO(takagi) other dtypes.
+    'use_cudnn': [True, False],
+    'use_outsize': [True, False],
+}))
+class TestDeconvolutionND(unittest.TestCase):
+
+    def setUp(self):
+        N = 2
+        in_channels = 3
+        out_channels = 2
+        ndim = len(self.dims)
+        ksize = (3,) * ndim
+        stride = (2,) * ndim
+        pad = (1,) * ndim
+
+        outs = tuple(
+            conv.get_deconv_outsize(d, k, s, p)
+            for (d, k, s, p) in zip(self.dims, ksize, stride, pad))
```

Comment:
I would like you to check the case in which outsize is determined by user, but its value is different from one determined by `get_deconv_outsize`.

When we fix the output size of convolution, there are several input sizes of the convolution that outputs the specified output size. (i.e. the equation get_conv_outsize(input_shape) == output_shape has multiple solutions of input_shape). So, when we make an autoencoder with a convolution and a deconvolution, we cannot determine the output size of the deconvolution solely from its input. This is why `Deconvolution2D` has `outsize` argument. `get_deconv_outsize` returns just one of the possible output size (maybe smallest or largest). So I would like you to check the behavior of `Deconvolution2D` with the condition mentioned above.